docs(how-tos): compute kubernetes CLI v2-MTA-1454 (#64)
scaleway/docs-content: compute/kubernetes/api-cli/creating-managing-kubernetes-lifecycle-cliv2.mdx

---
title: Creating and managing a Kubernetes Kapsule with CLI (v2)
description: This page explains how to create and manage a Kubernetes Kapsule with CLI (v2)
dates:
  validation: 2021-08-05
  posted: 2021-08-05
---

Scaleway's Kubernetes (K8s) Kapsule provides you with a managed environment to create, configure and run a cluster of pre-configured machines for containerized applications. You will be able to create clusters without the complexity of managing the infrastructure.

> You can [create](/compute/kubernetes/how-to/create-cluster/) and [manage](/compute/kubernetes/how-to/edit-cluster/) your Kapsule clusters from the console, via the API or via the [Scaleway Command Line Interface](https://developers.scaleway.com/en/products/k8s/api/).

Reviewer: ```suggestion
You can [create](/compute/kubernetes/how-to/create-cluster/) and [manage](/compute/kubernetes/how-to/edit-cluster/) your Kapsule clusters from the console, via the [API](https://developers.scaleway.com/en/products/k8s/api/) or via the [Scaleway Command Line Interface](https://developers.scaleway.com/en/products/k8s/api/).
```

  - You have generated your [API key](/console/my-project/how-to/generate-api-key/)
  - You have downloaded and installed the [Scaleway CLI](https://github.com/scaleway/scaleway-cli#scaleway-cli-v2)

</Message>

## Configuring the CLI

1. Open a terminal and type the following command, to configure the Scaleway CLI on your machine: 

  `scw init`
  
  If you have already configured the CLI, you will see a message like the following:

  ```
  Current config is located at /path/to/config.yaml
  access_key: <YOUR ACCESS KEY>
  secret_key: <YOUR SECRET KEY>
  default_organization_id: <YOUR ORGANIZATION ID>
  default_project_id: <YOUR PROJECT ID>
  default_region: <YOUR DEFAULT REGION, eg fr-par>
  default_zone: <YOUR DEFAULT ZONE, eg fr-par-1>

  Do you want to override the current config? (Y/n):  
  ```

2. Type `n` and go directly to the [Creating a K8s Kapsule cluster](#creating-a-k8s-kapsule-cluster) step of this tutorial, if you do not want to edit your configuration.

  If you have not yet configured the CLI, you will be guided through the configuration process via a series of questions:

  ```
  Enter a valid secret-key or an email
  ```

> 3. Enter either the email address for your account (you will then be asked for your password), or your [API Secret Key](/console/my-project/how-to/create-ssh-key/):

Reviewer: ```suggestion
3. Enter either the email address for your account (you will then be asked for your password), or your [API Secret Key](/console/my-project/how-to/generate-api-key/):
```

CreatedAt                     6 minutes ago
UpdatedAt                     now
Name                          new-pool
Status                        deleting
Version                       1.22.1
NodeType                      gp1_xs
Autoscaling                   false
Size                          5
MinSize                       0
MaxSize                       5
ContainerRuntime              containerd
Autohealing                   false
UpgradePolicy.MaxUnavailable  1
UpgradePolicy.MaxSurge        0
Zone                          fr-par-1
Region                        fr-par
```

Your pool has been deleted. 

## Uninstalling a Kubeconfig from a specific cluster 

Type the following command in your terminal and replace "your-cluster-ID" with the cluster ID you want to delete the Kubeconfig from:

`scw k8s kubeconfig uninstall your-cluster-ID` 

You will get an output similar to this: 

```
Cluster your-cluster-ID successfully deleted from /Users/yourusername/.kube/config
```

## Deleting a K8s cluster 

<Message type="important">

 This will permanently destroy your cluster and all associated pools and instances. 

</Message>

> Type the following command in your terminal and replace "your-cluster-ID" with the ID of the cluster you want to delete:

Reviewer 2: ```suggestion
Type the following command in your terminal and replace `your-cluster-ID` with the ID of the cluster you want to delete:
```

  DNSWildcard       *.12345678-1234-1234-4321-123456789123.nodes.k8s.fr-par.scw.cloud
  CreatedAt         now
  UpdatedAt         now
  DashboardEnabled  false
  Ingress           none
  UpgradeAvailable  false

  Autoscaler configuration:
  ScaleDownDisabled              false
  ScaleDownDelayAfterAdd         10m
  Estimator                      binpacking
  Expander                       random
  IgnoreDaemonsetsUtilization    false
  BalanceSimilarNodeGroups       false
  ExpendablePodsPriorityCutoff   -10
  ScaleDownUnneededTime          10m
  ScaleDownUtilizationThreshold  0.5
  MaxGracefulTerminationSec      600

  Auto-upgrade settings:
  Enabled                      false
  MaintenanceWindow.StartHour  0
  MaintenanceWindow.Day        any

  Open ID Connect configuration:
  IssuerURL       -
  ClientID        -
  UsernameClaim   -
  UsernamePrefix  -
  GroupsPrefix    -
  ```

  <Message type="note">

  For more information about the creation of a K8s cluster, type `scw k8s cluster create --help` in your terminal. 

  </Message>

2. Write down the ID of your cluster as you will need it in the next steps. 

> ## Installing a [KubeConfig](/compute/kubernetes/concepts/#kubeconfig/) configuration file 

Author: ```suggestion
## Installing a KubeConfig configuration file 
```

Type the following command in your terminal and replace "your-cluster-ID" with the ID of your previsouly-generated cluster: 

`scw k8s kubeconfig install you-cluster-ID`

You will see the following output: 

```
Kubeconfig for cluster your-cluster-ID successfully written at /Users/yourusername/.kube/config
```

> ## Adding a [pool](/compute/kubernetes/concepts/#pool/) to a K8s cluster 

Author: ```suggestion
## Adding a pool to a K8s cluster 
```

> ---

Reviewer: You need to add this doc to the side menu

You can create and manage your Kapsule clusters from the console, via the API or via the Scaleway Command Line Interface.

## Scaleway Command Line Interface Overview

[The Scaleway Command Line Interface (CLI)](https://github.com/scaleway/scaleway-cli) allows you to pilot your Scaleway infrastructure directly from your terminal, providing a faster way to administer and monitor your resources. Scaleway CLI is easy to set up and use, and an essential tool for operating efficiently in your cloud environment. The CLI provides many functionalities, including the ability to create and manage K8s Kapsules. 

<Message type="requirement">

  - You have an account and are logged into the [Scaleway Console](https://console.scaleway.com/)
  - You have generated your [API key](/console/my-project/how-to/generate-api-key/)
  - You have downloaded and installed the [Scaleway CLI](https://github.com/scaleway/scaleway-cli#scaleway-cli-v2)

>   - You have configured the [Scaleway CLI](/compute/bare-metal-cloud/api-cli/bare-metal-with-cli/#-Configuring-the-CLI)

Author: ```suggestion

```

</Message>

> ## Configuring the CLI

Reviewer 3: I don't think this should be in the tutorial because in the pre-requirements we suppose that the Scaleway CLI is configured.

  Do you want to send usage statistics and diagnostics? (Y/n):
  ```

4. Type `y` or `n` and hit enter.

  ```
  To fully enjoy Scaleway CLI we recommend you install autocomplete support in your shell.
  Do you want to install autocomplete? (Y/n):
  ```
5. Type `y` or `n` and hit enter.

  ```
  To enable autocomplete, scw needs to update your shell configuration.
  What type of shell are you using (default: zsh):
  ```

6. Hit enter if the default shell type is already correct for you. Otherwise, enter your shell type and hit enter.

  ```
  To enable autocomplete we need to append to /Users/yourusername/.zshrc the following lines:
  # Scaleway CLI autocomplete initialization.
  eval "$(scw autocomplete script shell=zsh)"
  Do you want to proceed with these changes? (Y/n):
  ```

7. Type `y` or `n` and hit enter. This should complete the configuration process:

  ```
  Initialization completed with success.
  ```

## Creating a K8s Kapsule cluster

1. Type the following command in your terminal to create a [K8S Kapsule cluster](/compute/kubernetes/concepts/#cluster/):

>   `scw k8s cluster create name=name-of-your-cluster version=1.22.1 pools.0.name=name-of-your-pool pools.0.node-type=GP1_XS pools.0.size=5`

Reviewer 3: You can skip the version, that way the command will continue to work even when 1.22.1 will be deprecated.

Reviewer 3: If you explain how to add a pool down below you can even simplify this command and just do: `scw k8s cluster create`

  DNSWildcard       *.12345678-1234-1234-4321-123456789123.nodes.k8s.fr-par.scw.cloud
  CreatedAt         now
  UpdatedAt         now
  DashboardEnabled  false
  Ingress           none
  UpgradeAvailable  false

  Autoscaler configuration:
  ScaleDownDisabled              false
  ScaleDownDelayAfterAdd         10m
  Estimator                      binpacking
  Expander                       random
  IgnoreDaemonsetsUtilization    false
  BalanceSimilarNodeGroups       false
  ExpendablePodsPriorityCutoff   -10
  ScaleDownUnneededTime          10m
  ScaleDownUtilizationThreshold  0.5
  MaxGracefulTerminationSec      600

  Auto-upgrade settings:
  Enabled                      false
  MaintenanceWindow.StartHour  0
  MaintenanceWindow.Day        any

  Open ID Connect configuration:
  IssuerURL       -
  ClientID        -
  UsernameClaim   -
  UsernamePrefix  -
  GroupsPrefix    -
  ```

  <Message type="note">

  For more information about the creation of a K8s cluster, type `scw k8s cluster create --help` in your terminal. 

  </Message>

2. Write down the ID of your cluster as you will need it in the next steps. 

> ## Installing a [KubeConfig](/compute/kubernetes/concepts/#kubeconfig/) configuration file 

Reviewer: You can't put links in headers unfortunately, it breaks the side menu (on the right). Same for the next header.

Type the following command in your terminal and replace "your-cluster-ID" with the ID of your previsouly-generated cluster: 

`scw k8s kubeconfig install you-cluster-ID`

You will see the following output: 

```
Kubeconfig for cluster your-cluster-ID successfully written at /Users/yourusername/.kube/config
```

## Adding a [pool](/compute/kubernetes/concepts/#pool/) to a K8s cluster 

> A pool is a set of identical Nodes. A pool has a name, a size (its current number of nodes), nodes number limits (min, max) and a Scaleway instance type. Changing those limits increases/decreases the size of a pool. Thus, when autoscaling is enabled, the pool will grow or shrink inside those limits, depending on its load. 

Reviewer 4: ```suggestion
A pool is a set of identical Nodes. A pool has a name, a size (its current number of nodes), nodes number limits (min, max), and a Scaleway instance type. Changing these limits increases/decreases the size of a pool. Thus, when autoscaling is enabled, the pool will grow or shrink inside those limits, depending on its load. 
```

Status            deleting
Version           1.22.1
Region            fr-par
OrganizationID    12345678-1234-1234-4321-123456789123
ProjectID         12345678-1234-1234-4321-123456789123
Cni               cilium
Description       -
ClusterURL        https://12345678-1234-1234-4321-123456789123.api.k8s.fr-par.scw.cloud:6443
DNSWildcard       *.12345678-1234-1234-4321-123456789123.nodes.k8s.fr-par.scw.cloud
CreatedAt         1 hour ago
UpdatedAt         now
DashboardEnabled  false
Ingress           none
UpgradeAvailable  false

Autoscaler configuration:
ScaleDownDisabled              false
ScaleDownDelayAfterAdd         10m
Estimator                      binpacking
Expander                       random
IgnoreDaemonsetsUtilization    false
BalanceSimilarNodeGroups       false
ExpendablePodsPriorityCutoff   -10
ScaleDownUnneededTime          10m
ScaleDownUtilizationThreshold  0.5
MaxGracefulTerminationSec      600

Auto-upgrade settings:
Enabled                      false
MaintenanceWindow.StartHour  0
MaintenanceWindow.Day        any

Open ID Connect configuration:
IssuerURL       -
ClientID        -
UsernameClaim   -
UsernamePrefix  -
GroupsPrefix    -
```

> 

Reviewer 3: remove useless whitelines here
